### PR TITLE
Fix x86_64-pc-linux-gnu and aarch64-pc-linux-gnu build

### PR DESCRIPTION
**Pull request by Blackhex**
Fixes build of x86_64-pc-linux-gnu and aarch64-pc-linux-gnu targets, contains changes from https://github.com/ZacWalk/gcc-woarm64/pull/5 that should be reviewed separately.

### CODE REVIEW

File: libgcc/config.host [hunk 1]
```diff
@@ -454,13 +454,16 @@ aarch64*-*-vxworks7*)
...
-aarch64-*-mingw*)
+aarch64*-*-mingw*)
 	case ${target_thread_file} in
 	  win32)
-	    tmake_file="$tmake_file i386/t-gthr-win32"
+	    tmake_thr_file="i386/t-gthr-win32"
 	    ;;
 	  posix)
-	    tmake_file="i386/t-mingw-pthread $tmake_file"
+	    tmake_thr_file="i386/t-mingw-pthread"
```
Comment (eukarpov) [inline libgcc/config.host]: please merge with the latest changes. are these changes really needed for posix and mcf now? is it tested somehow?

Comment (Blackhex) [reply libgcc/config.host]: Yes, as stated in the description, this had to be rebased on #5. Since #7 implemented more or less the same changes and it's merged now, this PR can be rebased (and already is) to the current `woarm64`branch.

**Review by eukarpov — APPROVED** [review]

File: gcc/common/config/aarch64/aarch64-common.cc [hunk 2]
```diff
@@ -426,6 +426,8 @@ aarch64_rewrite_mcpu (int argc, const char **argv)
   return aarch64_rewrite_selected_cpu (argv[argc - 1]);
 }
 
+#if TARGET_64BIT_MS_ABI
+
 /* Implement TARGET_EXCEPT_UNWIND_INFO.  */
 
 static enum unwind_info_type
```
Comment (eukarpov) [inline gcc/common/config/aarch64/aarch64-common.cc:433]: I am curious why unwind_info_type has not been defined earlier for aarch64

Comment (eukarpov) [reply gcc/common/config/aarch64/aarch64-common.cc:433]: This PR looks like a temporary solution for me to unblock validation for linux-gnu

**Comment by eukarpov** [comment]
CI check for this PR 
https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build-2/actions/runs/7020247752

**Review by eukarpov — COMMENTED** [review]
CI check for this PR
https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build-2/actions/runs/7020247752

**Review by ZacWalk — APPROVED** [review]
Tested. Looks good to me.
